pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -148,8 +148,10 @@ func (os *OpenStack) CreateVolume(name string, size int, tags *map[string]string
 	}
 
 	opts := volumes.CreateOpts{
-		Name: name,
-		Size: size,
+		Name:         name,
+		Size:         size,
+		VolumeType:   vtype,
```

Comment:
Have you run the e2e for provisioning + storage class with CINDER?  There's tests for CINDER but doesn't specify a volumeType.